source/MRMesh/MRObject.cpp

```diff
@@ -546,7 +551,7 @@ std::vector<std::string> Object::getInfoLines() const
     return res;
 }
 
-Expected<std::vector<std::future<Expected<void>>>> Object::serializeRecursive( const std::filesystem::path& path, Json::Value& root, int childId ) const
+Expected<std::vector<std::future<Expected<void>>>> Object::serializeRecursive( const std::filesystem::path& path, Json::Value& root, int childId, const std::filesystem::path& rootFolder, MapSharedObjectModelToLinkData* mapSharedObjectModelToLinkData ) const
```

Comment:
what if we move this implementation into protected `serializeRecursive_` function with input parameters structure and implement this one like this:
```c++
void serializeRecursive_(...)
{
    // current implementation
}

void serializeRecursive(  )
{
    // build map with links
    serializeRecursive_(); // will all extra parameters, so we don't need to change ObjectSave/ObjectLoad
}